### PR TITLE
[hotfix][docs] Review of concepts docs for grammar and clarity

### PR DESCRIPTION
**Pull request by ChrisChinchilla**
Spending some time doing a brief review of a few docs sections, just a start and I think more could be done.

I didn't change formatting or line breaks as much as I would like to as they're quite messy and inconsistent, but I'll save that for another time.

### CODE REVIEW

File: docs/concepts/programming-model.md [hunk 1]
```diff
@@ -33,53 +33,52 @@ Flink offers different levels of abstraction to develop streaming/batch applicat
...
+    The low level *Process Function* integrates with the *DataStream API*, making it possible to use the lower level abstraction
+    for certain operations. The *DataSet API* offers additional primitives on bounded data sets, like loops or iterations.
 
   - The **Table API** is a declarative DSL centered around *tables*, which may be dynamically changing tables (when representing streams).
-    The [Table API](../dev/table_api.html) follows the (extended) relational model: Tables have a schema attached (similar to tables in relational databases)
+    The [Table API](../dev/table_api.html) follows the (extended) relational model. Tables have a schema attached (similar to tables in relational databases)
     and the API offers comparable operations, such as select, project, join, group-by, aggregate, etc.
-    Table API programs declaratively define *what logical operation should be done* rather than specifying exactly
-   *how the code for the operation looks*. Though the Table API is extensible by various types of user-defined
+    Table API programs declaratively define *what logical operation should to perform* rather than specifying
```
Comment (zentol) [inline docs/concepts/programming-model.md]: remove "should"?

File: docs/concepts/runtime.md [hunk 2]
```diff
@@ -28,10 +28,10 @@ under the License.
 
 ## Tasks and Operator Chains
 
-For distributed execution, Flink *chains* operator subtasks together into *tasks*. Each task is executed by one thread.
-Chaining operators together into tasks is a useful optimization: it reduces the overhead of thread-to-thread
+For distributed execution, Flink *chains* operator subtasks together into *tasks*. One thread executes each task.
```
Comment (zentol) [inline docs/concepts/runtime.md]: hmm....i kind of prefer the old version.

File: docs/concepts/programming-model.md [hunk 3]
```diff
@@ -132,14 +131,13 @@ One typically distinguishes different types of windows, such as *tumbling window
 
 <img src="../fig/windows.svg" alt="Time- and Count Windows" class="offset" width="80%" />
 
-More window examples can be found in this [blog post](https://flink.apache.org/news/2015/12/04/Introducing-windows.html).
-More details are in the [window docs](../dev/stream/operators/windows.html).
+You can find more window examples in this [blog post](https://flink.apache.org/news/2015/12/04/Introducing-windows.html) and in the [window documentation](../dev/stream/operators/windows.html).
```
Comment (greghogan) [inline docs/concepts/programming-model.md]: Line break before "and".

File: docs/concepts/programming-model.md [hunk 4]
```diff
@@ -33,53 +33,52 @@ Flink offers different levels of abstraction to develop streaming/batch applicat
...
+    The [SQL](../dev/table_api.html#sql) abstraction closely interacts with the Table API, and you can execute SQL queries over tables defined in the *Table API*.
 
 
 ## Programs and Dataflows
 
-The basic building blocks of Flink programs are **streams** and **transformations**. (Note that the
-DataSets used in Flink's DataSet API are also streams internally -- more about that
-later.) Conceptually a *stream* is a (potentially never-ending) flow of data records, and a *transformation* is an
+The basic building blocks of Flink programs are **streams** and **transformations**. The
+DataSets used in Flink's DataSet API are also streams internally, which this document will cover later. Conceptually a *stream* is a (potentially never-ending) flow of data records, and a *transformation* is an
```
Comment (greghogan) [inline docs/concepts/programming-model.md]: Needs line break.

File: docs/concepts/runtime.md [hunk 5]
```diff
@@ -74,10 +74,10 @@ To control how many tasks a worker accepts, a worker has so called **task slots*
 Each *task slot* represents a fixed subset of resources of the TaskManager. A TaskManager with three slots, for example,
 will dedicate 1/3 of its managed memory to each slot. Slotting the resources means that a subtask will not
 compete with subtasks from other jobs for managed memory, but instead has a certain amount of reserved
-managed memory. Note that no CPU isolation happens here; currently slots only separate the managed memory of tasks.
+managed memory. No CPU isolation happens, slots only separate the managed memory of tasks.
```
Comment (greghogan) [inline docs/concepts/runtime.md]: "," -> ";"?

File: docs/concepts/runtime.md [hunk 6]
```diff
@@ -107,21 +107,20 @@ With hyper-threading, each slot then takes 2 or more hardware thread contexts.
...
+Programs written in the Data Stream API can resume execution from a **savepoint**. Savepoints allow both updating your programs and your Flink cluster without losing any state.
 
-[Savepoints](../ops/state/savepoints.html) are **manually triggered checkpoints**, which take a snapshot of the program and write it out to a state backend. They rely on the regular checkpointing mechanism for this. During execution programs are periodically snapshotted on the worker nodes and produce checkpoints. For recovery only the last completed checkpoint is needed and older checkpoints can be safely discarded as soon as a new one is completed.
+[Savepoints](../ops/state/savepoints.html) are **manually triggered checkpoints**, which take a snapshot of the program and write it out to a state backend. They rely on the regular checkpointing mechanism for this. During execution programs are periodically snapshotted on the worker nodes and produce checkpoints. The last completed checkpoint is only needed for recovery and you can safely discard older checkpoints as soon as a new one is completed.
```
Comment (greghogan) [inline docs/concepts/runtime.md]: Should we leave the original but change "can be safely" to "are" or "are by default"? The paragraph starts discussing savepoints but then switches to commenting on checkpoints with no distinction about externalized checkpoints.

**Comment by ChrisChinchilla** [comment]
@greghogan @zentol I will get these updates to you asap.

But I will have to strongly disagree on the 'you' and passive voice points. I've been working as a technical writer for some time, and this a point we're all fairly strong on pushing as in the long run it does make documents much clearer, less ambiguous and easier to read.

Granted a LOT of the Flink documentation doesn't follow this style right now, and that's something I personally would love to change over time, as I really think it would help people understand better.

Ideally I would love to create a more comprehensive style guide and explanation of this (and I have boilerplates I've used with other projects) and change this voicing over time.

Thoughts?

**Comment by ChrisChinchilla** [comment]
@greghogan That said, the whole 'we' vs 'you' discussion is purely a stylistic preference (if that's what you meant) as long as it's consistent and clear who you mean, but the comment about removing passive voice remains :)

File: docs/concepts/programming-model.md (hunk 4 above)
Comment (ChrisChinchilla) [reply docs/concepts/programming-model.md]: @greghogan I'm personally not a fan of hard line breaks, but happy to stick to them, but I'm struggling to figure out what the character limit for the project is, I can't see anything in any style files and it's somewhat varied throughout the docs. If there's a solid number then I'll add to the contributors guide.

File: docs/concepts/runtime.md (hunk 5 above)
Comment (ChrisChinchilla) [reply docs/concepts/runtime.md]: @greghogan With the full stop I don't think it's needed anymore.

**Comment by ChrisChinchilla** [comment]
Changes pushed.

**Comment by aljoscha** [comment]
I'm closing this as "Abandoned", since there is no more activity and the code base has moved on quite a bit. Please re-open this if you feel otherwise and work should continue.

**Pull request closed without merging**